src/candy-machine/candy-machine.service.ts

```diff
@@ -205,203 +238,275 @@ export class CandyMachineService {
-        collectionNftUri,
-        comicIssue.sellerFeeBasisPoints,
-      );
+      if (tokenStandard == TokenStandard.Core) {
+        const collection = generateSigner(umi);
+        const collectionBuilder = createCollectionV1(umi, {
+          collection,
+          uri: collectionNftUri,
+          name: onChainName,
+        });
+
+        await collectionBuilder.sendAndConfirm(umi, {
+          send: { commitment: 'confirmed' },
+        });
+
+        console.log(`Collection: ${collection.publicKey.toString()}`);
+        collectionNftAddress = new PublicKey(collection.publicKey);
+      } else {
+        const newCollectionNft = await createCollectionNft(
+          this.metaplex,
+          onChainName,
+          collectionNftUri,
+          comicIssue.sellerFeeBasisPoints,
+        );
+        collectionNftAddress = newCollectionNft.address;
+      }
+
       await this.prisma.collectionNft.create({
         data: {
-          address: newCollectionNft.address.toBase58(),
-          name: newCollectionNft.name,
+          address: collectionNftAddress.toBase58(),
+          name: onChainName,
           slug: uniqueSlug,
           comicIssue: { connect: { id: comicIssue.id } },
         },
       });
-      collectionNftAddress = newCollectionNft.address;
     }
 
-    const creators: CreateCandyMachineInput['creators'] = royaltyWallets.map(
-      (wallet) => ({
-        address: new PublicKey(wallet.address),
-        share: wallet.share,
-      }),
-    );
-
     const { startDate, endDate, mintLimit, freezePeriod, mintPrice, supply } =
       guardParams;
 
-    await initializeRecordAuthority(
-      this.metaplex,
-      candyMachineKey.publicKey,
-      collectionNftAddress,
-      new PublicKey(creatorAddress),
-      new PublicKey(creatorBackupAddress),
-      supply,
-    );
-
-    const groups: {
-      label: string;
-      guards: Partial<DefaultCandyGuardSettings>;
-    }[] = [
-      {
-        label: AUTHORITY_GROUP_LABEL,
-        guards: {
-          allowList: {
-            merkleRoot: getMerkleRoot([
-              this.metaplex.identity().publicKey.toString(),
-            ]),
-          },
-          solPayment: {
-            amount: solFromLamports(0),
-            destination: FUNDS_DESTINATION_ADDRESS,
-          },
-        },
-      },
-    ];
-
-    if (!!shouldBePublic) {
-      const paymentGuard = freezePeriod ? 'freezeSolPayment' : 'solPayment';
-      groups.push({
-        label: PUBLIC_GROUP_LABEL,
-        guards: {
-          startDate: { date: toDateTime(startDate) },
-          endDate: { date: toDateTime(endDate) },
-          [paymentGuard]: {
-            amount: solFromLamports(mintPrice),
-            destination: FUNDS_DESTINATION_ADDRESS,
-          },
-          mintLimit: mintLimit
+    let candyMachine: LegacyCandyMachine | CoreCandyMachine;
+    let candyMachineAddress: string;
+    if (tokenStandard == TokenStandard.Core) {
+      console.log('Create Core Candy Machine');
```

Comment:
don't want to repeat myself, but wondering if these console logs are intentional to keep it?